### ISSUE TITLE
Receive on browser

### ISSUE THREAD

**Issue by iFreilicht** [trebleshot/android#124]
# Problem

We already have **Send on browser** to allow sending files to devices where trebleshot is not installed or not even available to install. But, receiving files from such devices is currently not possible.

# Solution

Having a **Receive on brower** feature would solve this issue. It would be an additional option in the list of "options to connect to the other device". Like with **Send on browser**, an HTTP-server would be started and you would then see the IP and port of your Android device and a QR-code to connect to it from other devices. When doing so, an **Add files** button and/or drag-and-drop target would be shown, similar to cloud services and other websites that allow uploading files. Adding files and tapping a **Send** button would then send these files directly to the Android device.

# Alternatively: Use an app that can do this already

This feature is already implemented pretty well in [Fast File Transfer](https://play.google.com/store/apps/details?id=com.floriandraschbacher.fastfiletransfer). You have to pay to unlock this functionality, which I did. For the meantime, I will be using this app when I have to, but it is not open source.

Its workflow works like this:

## Select sending via browser

![Screenshot_20200625-163155](https://user-images.githubusercontent.com/9742635/85741602-dafc3200-b702-11ea-9cb8-b87684446157.png)

## Scan QR-Code on sending device

![screen2](https://user-images.githubusercontent.com/9742635/85741731-f5cea680-b702-11ea-9c53-dc55695abe07.png)

## Select files on sending device

![screen1](https://user-images.githubusercontent.com/9742635/85741756-fb2bf100-b702-11ea-8fdf-5a4d08bfef3b.png)

## Receive files as expected

![Screenshot_20200625-163320](https://user-images.githubusercontent.com/9742635/85741660-e64f5d80-b702-11ea-8523-98cfad23936c.png)

**Comment by mubashir-rehman** [comment]
This feature is already present under web share feature. Just need a little bit of rebranding. There is always a select file button when you open the link of trebleshot from any other device. It is there just not explained. Needs only optimization in terms of words to clear it properly.

**Comment by iFreilicht** [comment]
Ah nice, that's right!

I agree that it should be renamed, I'm not sure a lot of people actually know this is possible. Maybe all it needs is adding an option when tapping "Receive" that reads "Receive in browser"? Or the option in sidebar could be renamed "Host for browser"?

**Comment by velitasali** [comment]
I am not sure if we have an issue here. If the issue involves renaming a feature, could you open another issue for it? If what you are mentioning is unrelated, I will close this issue.

**Comment by iFreilicht** [comment]
The issue is that it is not clear that it's possible to receive files via browser the same way it is to send it. It is an existing but easy-to-miss feature IMO.

I think the fix may be two-fold:

- Maybe rename the "Share via browser" entry in the main hamburger menu to something that makes it clearer that you can both send and receive. Maybe just "Use browser"? This is debatable though.
- Add an additional entry called "Via browser" to the "Receive" menu. This way, this feature will be more clearly discoverable by users.

**Comment by velitasali** [comment]
Yes, I will move it from the navigation drawer (hamburger menu) to the navigation bar at the bottom. Though, I should say I haven't worked on that yet. 

For the second suggestion, it requires TrebleShot.  I think using the **Share via browser** option is unnecessary if both devices have TrebleShot installed.

**Comment by iFreilicht** [comment]
Nice, that would probably be a huge improvement already.

> I think using the Share via browser option is unnecessary if both devices have TrebleShot installed.

Yes, definitely 

> For the second suggestion, it requires TrebleShot.

I'm not sure what you mean? My idea was that on the receive menu (shown below), you'd also have an option "Via browser", so it's clear that the other person doesn't need TrebleShot to send you a file.

![Screenshot_20200924-165419](https://user-images.githubusercontent.com/9742635/94162444-fe13ba80-fe86-11ea-874e-f0f33ad8a6c1.png)

**Comment by velitasali** [comment]
> I'm not sure what you mean? My idea was that on the receive menu (shown below), you'd also have an option "Via browser", so it's clear that the other person doesn't need TrebleShot to send you a file.

Sorry, I have just understood what you meant. Yes, it could make sense to show internet addresses on that screen. I will work on this :+1:
